File: server/src/orders/models/order.model.ts
import {
  BelongsTo,
  Column,
  DataType,
  ForeignKey,
  Model,
  Table,
} from 'sequelize-typescript';
import { Expert } from 'src/experts/models/expert.model';
import { User } from 'src/users/models/user.model';

interface OrderCreationAttrs {
  object: string;
  object_type: 1 | 2;
  isDocs?: boolean;
  docsUrl?: string;
  status?: 'new' | 'inWork' | 'done';
}

@Table({ tableName: 'orders' })
export class Order extends Model<Order, OrderCreationAttrs> {
  @Column({
    type: DataType.INTEGER,
    autoIncrement: true,
    unique: true,
    primaryKey: true,
  })
  id: number;

  @Column({ type: DataType.STRING, allowNull: false, defaultValue: '' })
  object: string;

  @Column({ type: DataType.INTEGER, allowNull: false, defaultValue: 1 })
  object_type: 1 | 2;

  @Column({ type: DataType.BOOLEAN, allowNull: false, defaultValue: false })
  isDocs: boolean;

  @Column({ type: DataType.STRING, allowNull: false, defaultValue: '' })
  docs_url: string;

  @Column({ type: DataType.STRING, allowNull: false, defaultValue: 'new' })
  status: 'new' | 'inWork' | 'done';

  @ForeignKey(() => User)
  @Column({ type: DataType.INTEGER, unique: false})
  clientid: number

  @BelongsTo(() => User, { onDelete: 'CASCADE'})
  client: User;

  @ForeignKey(() => Expert)
  @Column({ type: DataType.INTEGER })
  expertid: number;

  @BelongsTo(() => Expert, { onDelete: 'CASCADE'})
  expert: Expert;
  
}

File: server/modules/orm/gorm/models/order.model.go
package model

import "time"

type Order struct {
	ID          uint64    `gorm:"primary_key, AUTO_INCREMENT"`
	Object      string    `json:"object"`
	Object_type int       `json:"object_type"`
	Address     string    `json:"address"`
	IsDocs      bool      `json:"is_docs"`
	DocsUrl     string    `json:"docs_url"`
	Status      string    `json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	ExpertID    *uint64   `gorm:"column:expert_id"`
	Expert      *User
	ClientID    *uint64 `gorm:"column:client_id"`
	Client      *User
}
